### PR TITLE
409422-connect-to-cosmos-snd4

### PR DESCRIPTION
**Pull request by simonnineteen90**
Co-authored-by: Amirs Ismuhametovs <JungaBunga@users.noreply.github.com>
Co-authored-by: AbidemiAdio <AbidemiAdio@users.noreply.github.com>

https://dev.azure.com/defragovuk/DEFRA-FFC/_boards/board/t/Single%20Front%20Door/Stories?System.Tags=SFD%20Tech&workitem=409422

### CODE REVIEW

File: .azuredevops/build.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ parameters:
 - name: deployFromFeature
   displayName: "Deploy from Feature Branch"
   type: boolean
-  default: false
+  default: true
```
Comment (rtasalem) [inline .azuredevops/build.yaml:5]: Ah is this why for some repos we have to go into the ADO pipeline to manually deploy from feature branch to SND environments? If so I'm assuming with each commit the build will automatically do the deployment from feature branch? 🤩

File: app/cosmos/client.js [hunk 2]
```diff
@@ -1,6 +1,11 @@
+const { DefaultAzureCredential } = require('@azure/identity')
 const { CosmosClient } = require('@azure/cosmos')
 const { cosmosConfig } = require('../config')
 
-if (cosmosConfig.key) {
+if (cosmosConfig.isDev || cosmosConfig.isTest) {
   module.exports = new CosmosClient({ endpoint: cosmosConfig.endpoint, key: cosmosConfig.key })
 }
+if (cosmosConfig.isProd) {
+  const credential = new DefaultAzureCredential({ managedIdentityClientId: cosmosConfig.managedIdentityClientId })
```
Comment (rtasalem) [inline app/cosmos/client.js:9]: Why do we only need the `DefaultAzureCredential` when the `cosmosConfig.isProd`, but not for when the `cosmosConfig.isDev` or `cosmosConfig.isTest`?

File: .azuredevops/build.yaml (hunk 1 above)
Comment (rtasalem) [reply .azuredevops/build.yaml:5]: Also on top of this question, guess it's worth keeping an eye out on future PRs across all our services for any repos where this is set to `false` and updating those to `true`?

Comment (simonnineteen90) [reply .azuredevops/build.yaml:5]: Yep thats correct! Agree we should have these set all to true. Its only on the oldest repos it was automated in new ones i think.

Comment (rtasalem) [reply .azuredevops/build.yaml:5]: Ahh cool, will keep an eye out then.

File: app/cosmos/client.js (hunk 2 above)
Comment (simonnineteen90) [reply app/cosmos/client.js:9]: In this scenario dev/test = local development so we need to use the Cosmos emulator which uses endpoint and key for connection.

Prod = anything deployed to Azure so SND and beyond and we need to use Managed Identity (which is the DefaultAzureCredential) to connect instead. We then have to make sure the service has the correct permissions to interact with cosmos as well. This is currently managed by ADP but they are introducing a change which will allow us to manage this through the infrastructure yaml

Comment (rtasalem) [reply app/cosmos/client.js:9]: Ahh nice, I do remember you saying before about still using the emulator for local dev so all makes sense now! Exciting!

**Review by JungaBunga — APPROVED** [review]
image

**Pull request merged**